core/src/cluster_nodes.rs

```diff
@@ -107,32 +108,42 @@ impl Node {
 
 impl<T> ClusterNodes<T> {
     pub(crate) fn submit_metrics(&self, name: &'static str, now: u64) {
+        let mut epoch_stakes = 0;
         let mut num_nodes_dead = 0;
         let mut num_nodes_staked = 0;
         let mut num_nodes_stale = 0;
+        let mut stake_dead = 0;
+        let mut stake_stale = 0;
         for node in &self.nodes {
+            epoch_stakes += node.stake;
             if node.stake != 0u64 {
                 num_nodes_staked += 1;
             }
             match node.contact_info() {
                 None => {
                     num_nodes_dead += 1;
+                    stake_dead += node.stake;
                 }
                 Some(node) => {
                     let age = now.saturating_sub(node.wallclock);
                     if age > CRDS_GOSSIP_PULL_CRDS_TIMEOUT_MS {
                         num_nodes_stale += 1;
+                        stake_stale += 1;
                     }
                 }
             }
         }
         num_nodes_stale += num_nodes_dead;
+        stake_stale += stake_dead;
```

Comment:
should this be `stake_stale_or_dead` ? Wasn't obvious to me at first glance that stale stake would be a superset of dead stake

Reply:
Well "dead" is effectively stalled for one+ epoch.
The `+=` is only because how the code is written:
https://github.com/solana-labs/solana/blob/d4b30adff/core/src/cluster_nodes.rs#L118-L126